ui/article-renderer/src/main/java/org/cru/godtools/article/activity/ArticlesActivity.java

```diff
@@ -57,14 +63,23 @@ protected void onStart() {
         loadInitialFragmentIfNeeded();
     }
 
+    @Override
+    public void onArticleSelected(@Nullable final Article article) {
+        Timber.tag("ArticlesActivity")
+                .d("Article selected: %s", article != null ? article.mTitle : null);
+    }
+
     // endregion Lifecycle Events
 
     private void loadInitialFragmentIfNeeded() {
         final FragmentManager fm = getSupportFragmentManager();
-
-        if (fm.findFragmentByTag(TAG_MAIN_FRAGMENT) == null) {
-           fm.beginTransaction().replace(R.id.articles_frame, ArticlesFragment
-                   .newInstance(mTool, mLocale, mCategoryId)).commit();
+        if (fm.findFragmentByTag(TAG_MAIN_FRAGMENT) != null) {
+            return;
         }
+
+        assert mTool != null : "if mTool was null the activity would have already finished";
+        fm.beginTransaction()
+                .replace(R.id.frame, ArticlesFragment.newInstance(mTool, mLocale, mCategoryId))
+                .commit();
```

Comment:
We didn't add the TAG in the replace call

Reply:
good catch!